File: bit/toHex.js
/**
 * @param {number} num
 * @return {string}
 */
const toHex = function(num) {
  let str = "";
  let n = num;
  let toHex; 
  let reminder;
  while(n > 0){
    n = num / 16;
    reminder = num % 16;
    if(reminder === 10){
      str += "a";
    }else if(reminder === 11){
      str += "b";
    }else if(reminder === 12){
      str += "c";
    }else if(reminder === 13){
      str += "d";
    }else if(reminder === 14){
      str += "e";
    }else if(reminder === 15){
      str += "f";
    }else{
      str += n.toString();
    }
  }
  return 
  
};

console.log(toHex(-1));

File: bit/ToHex.java
public class ToHex{
  public static String toHex(int num){

    if(num == 0){
      return "0";
    }

    String map ="0123456789abcdef";
    StringBuilder res = new StringBuilder();
    while(num != 0){
      res.insert(0, map.charAt(num & 0xF));
      num = num >>> 4;
    }  
    return res.toString();
  }

  public static void main(String[] args) {
    String ans = ToHex.toHex(-1);
    System.out.println(ans);
  }

}